contracts/CBOR.sol

```diff
@@ -52,6 +60,56 @@ library CBOR {
         buf.append(value);
     }
 
+    function encodeBigNum(Buffer.buffer memory buf, int value) internal pure {
+      buf.appendUint8(uint8((MAJOR_TYPE_TAG << 5) | TAG_TYPE_BIGNUM));
+      encodeBytes(buf, encodeUnsignedBigNum(uint(value)));
+    }
+
+    function encodeSignedBigNum(Buffer.buffer memory buf, int input) internal pure {
+      buf.appendUint8(uint8((MAJOR_TYPE_TAG << 5) | TAG_TYPE_NEGATIVE_BIGNUM));
+      encodeBytes(buf, encodeUnsignedBigNum(uint(-1 - input)));
+    }
+
+    function encodeUnsignedBigNum(uint value) internal pure returns (bytes) {
+      uint8 size = byteCount(value);
```

Comment:
Would it be valid CBOR to assume a max size and always use that, padded with zeros? I imagine we'd save on computation, but pay a small amount more on memory.

Reply:
I don't think so, maybe a little antithetical to the saving storage philosophy of CBOR, but not against the spec.